validation.go

```diff
@@ -53,7 +74,21 @@ func (v Validator) VerifyRecord(r *pb.Record) error {
 		return ErrInvalidRecordType
 	}
 
-	return val.Func(r.GetKey(), r.GetValue())
+	author := peer.ID("")
+	if len(r.GetSignature()) > 0 {
```

Comment:
let's add a comment here that the signature has been validated by the DHT; it's critical to be clear about this imo.

Reply:
@vyzo Yes I agree it's hard to follow where all the pieces fit together. In my opinion it would be better for signature validation to happen inside this module, perhaps by passing a public key into the verify function. But it would need some rearchitecting because the signature is optional

Reply:
yeah, no need to complicate. A comment (and perhaps a TODO) will do for now.